index.js

```diff
@@ -243,6 +253,56 @@ module.exports = function(version, _options) {
             }
 
             return output;
+        },
+        getBestMatchingLanguage: function(language) {
+            if (languages.instructions[language]) return language;
+
+            // If the language is not found, try a little harder
+            var splitLocale = language.toLowerCase().split('-');
+            var languageCode = false;
+            var scriptCode = false;
+            var countryCode = false;
+
+            if (splitLocale.lenth === 1) {
+                languageCode = splitLocale[0];
+            } else if (splitLocale.length === 2 && splitLocale[1].length === 4) {
+                languageCode = splitLocale[0];
+                scriptCode = splitLocale[1];
+            } else if (splitLocale.length === 2 && splitLocale[1].length === 2) {
+                languageCode = splitLocale[0];
+                countryCode = splitLocale[1];
+            } else if (splitLocale.length === 3) {
+                languageCode = splitLocale[0];
+                scriptCode = splitLocale[1];
+                countryCode = splitLocale[2];
+            }
+
+            var supportedLanguageCode = languages.supportedCodes.map(function(locale) {
+                return locale.toLowerCase().split('-')[0];
+            });
+
+            // Same language code and script code (lng-Scpt)
+            if (languages.instructions[languageCode + '-' + scriptCode]) {
+                return languageCode + '-' + scriptCode;
+
+            // Same language code and country code (lng-CC)
+            } else if (languages.instructions[languageCode + '-' + countryCode]) {
+                return languageCode + '-' + countryCode;
+
+            // Same language code (lng)
+            } else if (supportedLanguageCode.indexOf(languageCode) > -1) {
+                return languages.supportedCodes[supportedLanguageCode.indexOf(languageCode)];
+
+            // Same language code and any script code (lng-Scpx)
+            } else if (supportedLanguageCode.indexOf(languageCode) > -1 && scriptCode) {
```

Comment:
If we support `zh-Hans` but not `zh`, and `language` is `zh-Hant`, then we won’t go into this `if` statement because `zh-Hant` isn’t one of the supported languages. Instead, we need to look for an item in `supportedLanguageCodes` that contains the language code `zh` and also contains a script code.

Reply:
Addressed in https://github.com/Project-OSRM/osrm-text-instructions/pull/168/commits/167f7c2e783d043d6bce307f4173d16115b7f0f0